master/internal/api_workspace.go

```diff
@@ -368,10 +409,36 @@ func (a *apiServer) PostWorkspace(
 		return nil, errors.Wrapf(err, "error creating workspace %s in database", req.Name)
 	}
 
+	// Verify that the specified cluster name is also provided in the master config.
+	if req.ClusterName != nil {
+		newReq := &apiv1.ModifyWorkspaceNamespaceBindingRequest{
+			Id:                  int32(w.ID),
+			ClusterName:         *req.ClusterName,
+			NamespaceName:       req.NamespaceName,
+			AutoCreateNamespace: req.AutoCreateNamespace,
+		}
+
+		_, err := a.modifyWorkspaceNamespaceBinding(ctx, newReq, &tx, w)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to create namespace binding: %w", err)
+		}
+		if req.Quota != nil {
+			quotaReq := &apiv1.SetResourceQuotaRequest{
+				Quota:       *req.Quota,
+				ClusterName: *req.ClusterName,
+				Id:          int32(w.ID),
+			}
+			_, err = a.setResourceQuota(ctx, quotaReq, &tx, w)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to set the resource quota")
```

Comment:
blocker: replace all instances of errors.Wrap with fmt.Errorf (https://hpe-aiatscale.atlassian.net/wiki/spaces/ENGINEERIN/pages/1389035546/Go+Style+at+Determined#Error-Wrapping)